server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -687,6 +688,10 @@ public Discoverer getMatchingDiscover(final Hypervisor.HypervisorType hypervisor
         if ((clusterName != null || clusterId != null) && podId == null) {
             throw new InvalidParameterValueException("Can't specify cluster without specifying the pod");
         }
+        if (!HypervisorType.VMware.toString().equalsIgnoreCase(hypervisorType) &&
+                (Strings.isNullOrEmpty(username) || Strings.isNullOrEmpty(password))) {
```

Comment:
We could import `org.apache.commons.lang3.StringUtils` here and use `StringUtils.isAnyEmpty(username, password)`.

Reply:
I agree with @GutoVeronezi.

Nowadays we already have some discussions about adopting `org.apache.commons.lang3.StringUtils` vs `com.cloud.utils.StringUtils` "_facade_" (@DaanHoogland :slightly_smiling_face:); with the `google.common.base.Strings` adds a third variable to this "equation".